aws/table_aws_identitystore_group.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/identitystore"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsIdentityStoreGroup(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_identitystore_group",
+		Description: "AWS Identity Store Group",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"identity_store_id", "id"}),
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceNotFoundException"}),
+			Hydrate:           getIdentityStoreGroup,
+		},
+		List: &plugin.ListConfig{
+			KeyColumns: plugin.AllColumns([]string{"identity_store_id", "name"}),
```

Comment:
Hi @LalitTurbot 

Whilst the API docs might show it as optional, if you make a call to ListGroups without specifying any filter you'll get an error back:

```
% aws identitystore list-groups --identity-store-id d-1234567890 

An error occurred (ValidationException) when calling the ListGroups operation: Filter is required
```

Reply:
Ah - the docs also state Filter is required: https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/API_ListGroups.html

> Lists the attribute name and value of the group that you specified in the search. We only support DisplayName as a valid filter attribute path currently, and **filter is required**.

Reply:
The use case here isn't to list groups specifically (or not yet anyway, not until the API supports that in a future version), but more to join to the list of SSO permission set/account/group assignments (that only has a group ID, so joining to this table would allow returning the actual group names)